kyaml/kio/filters/container.go

```diff
@@ -408,52 +407,67 @@ const (
 
 var functionAnnotationKeys = []string{FunctionAnnotationKey, oldFunctionAnnotationKey}
 
-// GetFunction parses the config function from the object if it is found
-func GetFunction(n *yaml.RNode, meta yaml.ResourceMeta) (*yaml.RNode, error) {
+// getFunction parses the config function from the object if it is found
+func getFunction(n *yaml.RNode, meta yaml.ResourceMeta) *FunctionSpec {
+	var fs FunctionSpec
 	for _, s := range functionAnnotationKeys {
 		fn := meta.Annotations[s]
 		if fn != "" {
-			return yaml.Parse(fn)
+			_ = yaml.Unmarshal([]byte(fn), &fs)
+			return &fs
 		}
 	}
-	return n.Pipe(yaml.Lookup("metadata", "configFn"))
+	n, err := n.Pipe(yaml.Lookup("metadata", "configFn"))
+	if err != nil || yaml.IsEmpty(n) {
+		return nil
+	}
+	s, err := n.String()
+	if err != nil {
+		return nil
+	}
+	_ = yaml.Unmarshal([]byte(s), &fs)
+	return &fs
 }
 
-// GetContainerName returns the container image for an API if one exists
-func GetContainerName(n *yaml.RNode) (string, string) {
-	meta, _ := n.GetMeta()
-
-	// path to the function, this will be mounted into the container
-	path := meta.Annotations[kioutil.PathAnnotation]
-
-	fn, _ := GetFunction(n, meta)
-	if fn != nil {
-		image, _ := fn.Pipe(yaml.Lookup("container", "image"))
-		return yaml.GetValue(image), path
-	}
+type ContainerSpec struct {
```

Comment:
Are these the only 2 flags we support ?  Will there be any more of these?

Reply:
yes to both